src/plugins/intel_cpu/src/transformations/cpu_opset/x64/pass/convert_precision_i64_i32.cpp

```diff
@@ -0,0 +1,97 @@
+    const auto* srcData = constant->get_data_ptr<int64_t>();
+    const auto size = shape_size(constant->get_shape());
+
+    auto newConstant = std::make_shared<op::v0::Constant>(element::i32, constant->get_shape());
+    newConstant->output(0).set_names(constant->output(0).get_names());
+    auto* dstData = const_cast<int32_t *>(reinterpret_cast<const int32_t *>(newConstant->get_data_ptr()));
+    OPENVINO_ASSERT(dstData != nullptr, "Can't get destination data pointer");
+
+    for (size_t i = 0; i < size; ++i) {
+        if (srcData[i] >= std::numeric_limits<int32_t>::max()) {
+            dstData[i] = std::numeric_limits<int32_t>::max();
+        } else if (srcData[i] <= std::numeric_limits<int32_t>::lowest()) {
+            dstData[i] = std::numeric_limits<int32_t>::lowest();
+        } else {
+            dstData[i] = static_cast<int32_t>(srcData[i]);
+        }
+    }
+    return newConstant;
+}
+
+bool ConvertPrecisionI64ToI32::run_on_model(const std::shared_ptr<ov::Model>& model) {
+    const auto orderedOps = model->get_ordered_ops();
+    for (const auto& op : orderedOps) {
+        if (isNativelySupported(op->get_type_info()) || TypeFromName(op->get_type_name()) == Type::Unknown) {
+            continue;
+        }
+
+        bool convertForOutputsRequired = false;
+        for (const auto& input : op->inputs()) {
+            if (input.get_element_type() == element::i64) {
+                auto parentOutput = input.get_source_output();
+                auto parentNode = parentOutput.get_node_shared_ptr();
+                if (is_type<opset12::Convert>(parentNode) &&
+                        parentNode->get_input_element_type(0) == element::i32 &&
+                        parentNode->get_output_element_type(0) == element::i64) {
+                    input.replace_source_output(parentNode->input_value(0));
+                } else if (auto constOp = as_type_ptr<op::v0::Constant>(parentNode)) {
+                    auto newConst = changeConstantPrecision(constOp);
+                    input.replace_source_output(newConst);
+                    newConst->set_friendly_name(constOp->get_friendly_name());
+                } else {
+                    auto convert = std::make_shared<opset12::Convert>(input.get_source_output(), element::i32);
+                    convert->output(0).add_names(parentOutput.get_names());
+                    input.replace_source_output(convert);
+                }
+                convertForOutputsRequired = true;
+            }
+        }
+
+        if (convertForOutputsRequired) {
+            // Propagate i32 precision into outputs.
+            op->validate_and_infer_types();
+            for (auto& output : op->outputs()) {
+                if (output.get_element_type() == element::i32) {
+                    auto convert = std::make_shared<opset12::Convert>(output, element::i64);
+                    replace_output_update_name(output, convert->input_value(0));
```

Comment:
`output` and `convert->input_value(0)` are the same. I suppose in this case we need to replace `output` with convert's output. BTW do we cover this pass by tests (e.g. by existing tests)? If no, can we add them?
 ```suggestion
                    replace_output_update_name(output, convert->output(0));
```

Reply:
Fixed. Test was added